pandas/core/arrays/boolean.py

```diff
@@ -126,7 +126,12 @@ def __from_arrow__(
             bool_arr = BooleanArray._from_sequence(np.array(arr))
             results.append(bool_arr)
 
-        return BooleanArray._concat_same_type(results)
+        if len(results) == 0:
```

Comment:
not results here instead of length check?

Reply:
What would be the reasoning for this? `== 0` is the more readable version IMO.

Reply:
Consistency, I think we are mostly using not result in these cases, but might be wrong here

Reply:
Updated and CI passed except for the `ResourceWarning`.